frontend/src/app/supportAdmin/UnarchivePatients/UnarchivePatient.tsx

```diff
@@ -0,0 +1,425 @@
+      let res = await queryGetOrgWithFacilities({
+        variables: { id: selectedOrgInternalId },
+      });
+      let facilities = res.data?.organization?.facilities
+        ? res.data?.organization?.facilities
+        : [];
+      if (facilities !== undefined && facilities.length > 0) {
+        facilities.sort((a, b) => (a.name > b.name ? 1 : -1));
+      }
+      setFacilities(facilities);
+    }
+  };
+
+  const onFacilityChange = (facilityId: string) => {
+    if (facilityId === "") {
+      setArchivedPatientsCount(undefined);
+      setArchivedPatients(undefined);
+    }
+    setSelectedFacilityId(facilityId);
+  };
+
+  const fetchAndSetArchivedPatientsCount = async (
+    orgInternalId: string,
+    facilityId: string,
+    organizations: UnarchivePatientOrganization[]
+  ) => {
+    if (orgInternalId && facilityId && organizations) {
+      setIsLoadingCount(true);
+      let orgExternalId = getOrgExternalIdWithInternalId(
+        organizations,
+        orgInternalId
+      );
+      if (orgExternalId) {
+        let { data } = await queryGetPatientsCountByFacilityWithOrg({
+          variables: {
+            facilityId: facilityId,
+            archivedStatus: ArchivedStatus.Archived,
+            orgExternalId: orgExternalId,
+          },
+        });
+        let newArchivedPatientsCount = data?.patientsCount || 0;
+        setArchivedPatientsCount(newArchivedPatientsCount);
+      }
+      setIsLoadingCount(false);
+    }
+  };
+
+  const fetchAndSetArchivedPatients = async (
+    orgInternalId: string,
+    facilityId: string,
+    organizations: UnarchivePatientOrganization[]
+  ) => {
+    if (orgInternalId && facilityId && organizations) {
+      setIsLoadingPatients(true);
+      let orgExternalId = getOrgExternalIdWithInternalId(
+        organizations,
+        orgInternalId
+      );
+      if (orgExternalId) {
+        let { data } = await queryGetPatientsByFacilityWithOrg({
+          variables: {
+            facilityId: facilityId,
+            pageNumber: currentPage - 1,
+            pageSize: entriesPerPage,
+            archivedStatus: ArchivedStatus.Archived,
+            orgExternalId: orgExternalId,
+          },
+        });
+        let newArchivedPatients: any[] = data?.patients || [];
+        setArchivedPatients(newArchivedPatients);
+        setIsLoadingPatients(false);
+      }
+    }
+  };
+
+  const onSubmit = async () => {
+    const values = getValues();
+    let orgId = values.organization;
+    let facilityId = values.facility;
+    if (orgId && facilityId) {
+      let searchParams = "";
+      if (activeFacility?.id) {
+        searchParams = `?facility=${activeFacility.id}`;
+      }
+      await fetchAndSetArchivedPatients(orgId, facilityId, organizations);
+      await fetchAndSetArchivedPatientsCount(orgId, facilityId, organizations);
+      // clear pagination on search submit
+      navigate(`${unarchivePatientPageUrl}${searchParams}`);
```

Comment:
⚠️ not sure if this is too hacky...
But our pagination component adds `/{number}` to the end of the path so if someone was on the 3rd page of results their path would be updated to `/admin/unarchive-patient/3`

At that point if they select a new facility and new org and click "Search" and that result does not have 3 pages of results it will display "No result" which I found confusing.

The only downside is the get patients query runs 2x in the above scenario (one with the page number of 2 and one with page number of 0)

Let me know if I need to elaborate more 😅